web3c/index.js

```diff
@@ -38,15 +38,14 @@ if (typeof Web3 !== 'undefined' && (new Web3()).version && !(new Web3()).version
   localWeb3 = Web3;
   resolveWeb3(module.exports);
 // Option 2: At load time, the webpack module does not find web3.
-// Require.ensure allows loading the bundled, compiled version of web3 as
-// as a separate browser request, with the downside that it is asynchronous,
-// and means that `new web3c()` will not be functional immediately.
+// import loads the bundled, compiled version of web3 as a separate
+// browser request, with the downside that it is asynchronous, and
+// means that `new web3c()` will not be functional immediately.
 } else if (typeof define !== 'undefined') {
-  // webpack
-  require.ensure(['web3'], function(require) {
-    localWeb3 = require('web3');
+  import('web3').then(_web3 => {
```

Comment:
React complains about `require.ensure`. Hence the change to import and all of the concomitant changes to update babel and webpack, as a result.

Reply:
`import()` is apparently now suggested over `require.ensure`. See https://webpack.js.org/guides/code-splitting/#dynamic-imports

Reply:
probably want a `/* webpackChunkName: "web3" */` in the import to hint at the name for the chunk?

Reply:
since this file is where we're probing support of the environment, can we support both options here?
```
} else if (typeof define !== 'undefined' && typeof import !== 'undefined') {
  import('web3')...
} else if (typeof define !== 'undefined') {
  require.ensure(['web3'],...)
}
```

if we can gate the code without too much work so it'll work in a legacy webpack project that doesn't put in the babel transforms / polyfills, that seems like not a bad thing

Reply:
seems like a good idea

Reply:
https://stackoverflow.com/questions/40329816/detect-if-es2015-code-was-transpiled-by-babel-or-is-running-native

Reply:
you'll still run into babel-webpack not knowing how to transpile the `require.ensure` though

Reply:
well that sure looks robust and definitely not error prone :P